app/controllers/concerns/api/v3/sync_to_user.rb

```diff
@@ -32,10 +32,13 @@ def processed_until(records)
     end
 
     def response_process_token
-      {current_facility_id: current_facility.id,
-       current_facility_processed_since: processed_until(current_facility_records) || current_facility_processed_since,
-       other_facilities_processed_since: processed_until(other_facility_records) || other_facilities_processed_since,
-       resync_token: resync_token}
+      {
+        current_facility_id: current_facility.id,
+        current_facility_processed_since: processed_until(current_facility_records) || current_facility_processed_since,
+        other_facilities_processed_since: processed_until(other_facility_records) || other_facilities_processed_since,
+        resync_token: current_resync_token,
+        sync_region_id: current_sync_region_id
```

Comment:
```suggestion
        sync_region_id: sync_region.id
```

Reply:
Case where this change might break: 
- We enable block level sync for a user, they haven't resynced facilities yet. The `SYNC_REGION_ID` header is set to the facility group.
- App syncs and starts receiving blocks in the response. Since the `process_token.sync_region_id` is blank on first call, resync is triggered. Also `process_token.sync_region_id` is set to Block region's ID.
- They will send the same old `SYNC_REGION_ID` in the next request, we trigger a resync again.

If we mirror the header in `process_token.sync_region_id` (currently this way) there's a difference between what we are sending (block level data) and what the `process_token.sync_region_id` represents (facility group data).

Reply:
This has been addressed by making `current_sync_region` sensitive to the various possible `sync_region_id`'s.